### PR TITLE
Added a state machine for cache partition.

### PR DESCRIPTION
**Pull request by vldpyatkov**
Now two operation is available over distributed table. There are insert operation and get operation here.

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/GetCommand.java [hunk 1]
```diff
@@ -1,12 +1,73 @@
...
 package org.apache.ignite.internal.table.distributed.command;
 
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.schema.ByteBufferRow;
+import org.apache.ignite.lang.IgniteLogger;
 import org.apache.ignite.raft.client.ReadCommand;
 
 public class GetCommand implements ReadCommand {
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/GetCommand.java:27]: serialversionuid

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/InsertCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+package org.apache.ignite.internal.table.distributed.command;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.schema.ByteBufferRow;
+import org.apache.ignite.lang.IgniteLogger;
+import org.apache.ignite.raft.client.WriteCommand;
+
+public class InsertCommand implements WriteCommand {
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/InsertCommand.java:27]: serialversionuid

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/response/KVGetResponse.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.schema.ByteBufferRow;
+import org.apache.ignite.internal.table.distributed.command.GetCommand;
+import org.apache.ignite.lang.IgniteLogger;
+
+/**
+ * It is a response object for handling a table get command.
+ * {@see org.apache.ignite.internal.table.distributed.command.GetCommand}
+ */
+public class KVGetResponse implements Serializable {
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/response/KVGetResponse.java:32]: serialversionuid

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/response/KVInsertResponse.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+package org.apache.ignite.internal.table.distributed.command.response;
+
+import java.io.Serializable;
+
+/**
+ * It is a response object for handling a table insert command.
+ * {@see org.apache.ignite.internal.table.distributed.command.InsertCommand}
+ */
+public class KVInsertResponse implements Serializable {
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/response/KVInsertResponse.java:26]: serialversionuid

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            assert clo.command() instanceof InsertCommand;
+
+            BinaryRow previous = storage.putIfAbsent(
+                extractAndWrapKey(((InsertCommand)clo.command()).getRow()),
+                ((InsertCommand)clo.command()).getRow()
+            );
+
+            clo.success(new KVInsertResponse(previous == null));
+        }
+
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java:66]: blank line

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onWrite(Iterator<CommandClosure<WriteCommand>> iterator) {
+        while (iterator.hasNext()) {
+            CommandClosure<WriteCommand> clo = iterator.next();
+
+            assert clo.command() instanceof InsertCommand;
+
+            BinaryRow previous = storage.putIfAbsent(
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java:59]: Why it's putIfAbsent? Should we have an ability to rewrite value?

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                extractAndWrapKey(((InsertCommand)clo.command()).getRow()),
+                ((InsertCommand)clo.command()).getRow()
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java:61]: Why to store key both as key and as part of value?

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    private HashMap<KeyWrapper, BinaryRow> storage = new HashMap<>();
+
+    /** {@inheritDoc} */
+    @Override public void onRead(Iterator<CommandClosure<ReadCommand>> iterator) {
+        while (iterator.hasNext()) {
+            CommandClosure<ReadCommand> clo = iterator.next();
+
+            assert clo.command() instanceof GetCommand;
+
+            clo.success(new KVGetResponse(storage.get(extractAndWrapKey(((GetCommand)clo.command()).getKeyRow()))));
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java:48]: Do we need exception handling here?

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java (hunk 5 above)
Comment (vldpyatkov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java:66]: Done.

**Pull request closed without merging**